modules/hal_nordic/nrfx/Kconfig

```diff
@@ -129,7 +129,17 @@ config NRFX_PWM3
 
 config NRFX_QDEC
 	bool "Enable QDEC driver"
-	depends on HAS_HW_NRF_QDEC
+	depends on HAS_HW_NRF_QDEC || HAS_HW_NRF_QDEC0 || HAS_HW_NRF_QDEC1
+
+config NRFX_QDEC0
+	bool "Enable QDEC0 instance"
+	depends on HAS_HW_NRF_QDEC || HAS_HW_NRF_QDEC0
+	select NRFX_QDEC
+
+config NRFX_QDEC1
+	bool "Enable QDEC1 instance"
```

Comment:
As I mentioned above, only `NRF_QDEC`/`NRF_QDEC0` is currently supported by the driver, so the user cannot actually select instances to be enabled. So at this point I think it's undesirable to introduce these two options.